tools/chplcheck/src/rules.py

```diff
@@ -375,8 +393,13 @@ def UnusedFormal(_, root: AstNode):
             if refersto:
                 uses.add(refersto.unique_id())
 
+        lines = context.get_file_text(root.location().path()).split("\n")
         for unused in formals.keys() - uses:
-            yield AdvancedRuleResult(formals[unused], formals[unused].parent())
+            anchor = formals[unused].parent()
+            ignore = build_ignore_fixit(anchor, lines, UnusedFormal.__name__)
+            yield AdvancedRuleResult(
+                formals[unused], anchor, fixit=[ignore]
+            )
```

Comment:
Looks like advanced results should know exactly where to put the auto-fixit for "ignore" because they get the anchor. Suggest you move that logic into `AdvancedResult` to auto-handle that. Perhaps you need to adjust the `.fixits` to be a getter that takes a context (so that you can get lines)

Reply:
In the current setup, ignore fixits are treated the same as any other kind of fixit. If I understand your suggestion right, you are proposing something like:

```python3
yield AdvancedRuleResult(
    formals[unused], anchor, ignorable=True 
)
```

and for basic rules:

```python3
return BasicRuleResult(node, ignorable=True)
```

Reply:
Even better -- the only purpose for `anchor`s is to allow silencing linter rules. Thus, if an advanced rule allows for anchors, it allows its warnings to be silenced using attributes. Hence, for `AdvancedRuleResult`, having an anchor should be enough to auto-add an 'ignore' fixit.

For basic rules, a boolean argument might be needed.

Reply:
I have implemented this